### PR TITLE
[cu-30xvvc2] Remove CurrencyPair use in pool data structure

### PR DESCRIPTION
**Pull request by vimukthi-git**
## Issue
Refer issue

### CODE REVIEW

**Comment by ComposableFiB0t** [comment]
Task linked: [CU-30xvvc2 5.3) 1. Implement Refactoring: CurrencyPair Usage.](https://app.clickup.com/t/30xvvc2)

File: code/parachain/frame/pablo/src/dual_asset_constant_product.rs [hunk 1]
```diff
@@ -60,44 +64,54 @@ impl<T: Config> Uniswap<T> {
...
+			compute_out_given_in(quote_asset.1, base_asset.1, quote_asset.2, base_asset.2, amount)
 		} else {
-			compute_in_given_out(
-				pool.quote_weight,
-				pool.base_weight,
-				pool_quote_aum,
-				pool_base_aum,
-				amount,
-			)
+			compute_in_given_out(quote_asset.1, base_asset.1, quote_asset.2, base_asset.2, amount)
```
Comment (PoisonPhang) [inline code/parachain/frame/pablo/src/dual_asset_constant_product.rs:90]: Time to upgrade to new methods now that they are in main?

Comment (vimukthi-git) [reply code/parachain/frame/pablo/src/dual_asset_constant_product.rs:90]: That comes in a later task

**Review by hussein-aitlahcen — APPROVED** [review]

**Review by PoisonPhang — APPROVED** [review]

**Review by Martin-Kiss — APPROVED** [review]

**Review by Dom-Roth — APPROVED** [review]

**Comment by vimukthi-git** [comment]
@mergifyio refresh

**Comment by vimukthi-git** [comment]
@Mergifyio refresh

**Pull request merged**